### PR TITLE
Issue #388/389: Update developer guide and PPP

### LINKED ISSUE #388: Update DG - Add manual test cases, use cases and user stories for `remark`, `sort`, `find` and `view`

### LINKED ISSUE #389: Update PPP

### PR DESCRIPTION
**Pull request by charmainehly**
Details of PR: Adds manual test cases, use cases and user stories to the developer guide for the `remark`, `view`, `sort` and `find` command. (pending implementation details)

Note: this commit also updates details for the PPP

Closes #388 
Closes #389

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -516,6 +516,8 @@ that this will be a heavy headway for our application.
 The following activity diagram summarizes what happens when a user executes a `focus` command.<br>
 <img src="images/FocusCommandDiagram.png" width="250" />
 
+
+#TO DELETE ****
```
Comment (domlimm) [inline docs/DeveloperGuide.md:520]: @charmainehly might need to add a spacing for the header and to appear in DG's TOC. Feel free to just commit and resolve via the button for convenience!

```suggestion
# TO DELETE ****
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -619,34 +621,33 @@ _{Explain here how the data archiving feature will be implemented}_
 **Target user profile**:
 
 * Professor from NUS School of Computing
-* Needs to manage a significant number of TA applications
+* Needs to manage a significant number of TA applications for a particular module
```
Comment (domlimm) [inline docs/DeveloperGuide.md:624]: Our app. is missing the module part where we did not take into account the interview is for which module 😢 

What's your thoughts on the updated Line 624?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -619,34 +621,33 @@ _{Explain here how the data archiving feature will be implemented}_
...
 * Prefers typing to mouse interactions
 * Reasonably comfortable using CLI apps
 
-**Value proposition**: TAlent Assistant™ creates a centralized management system for professors to manage TA
-applications by providing access to various application data such as qualifications, experience and
-interview availability dates.
-
+**Value proposition**: TAlent Assistant™ creates a centralized management system for NUS School of Computing professors to manage 
+undergraduate TA applications by providing easy access to candidates' data to review their general availability for
+scheduling interviews during office hours.
```
Comment (domlimm) [inline docs/DeveloperGuide.md:632]: Nice! 👍🏻

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -619,34 +621,33 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* * *`  | user     | find a candidate in the system                          | access details of the candidate without having to go through the entire list.      |
+| `* * *`  | user     | list all candidates in the system                       | monitor the application pool.                                                      |
+| `* *`    | user     | sort candidates in the system                           | view the candidates in a more organised manner based on a certain attribute field. |
+| `* * *`  | user     | view scheduled interviews within a specific time period | keep track of the upcoming interview schedule.                                     |
+| `* * *`  | user     | schedule TA candidates for interviews                   | keep track of the interview schedule.                                              |
+| `* * *`  | user     | update the remark for a candidate in the system         | keep a note of important details relating to the candidate.                        |
+| `* * *`  | new user | view all available commands                             | get familiarised with the system.                                                  |
```
Comment (domlimm) [inline docs/DeveloperGuide.md:649]: I'll work on the DG later. Will check if we're missing any user stories and then add it in 👍🏻

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -910,13 +943,13 @@ testers are expected to do more *exploratory* testing.
...
 
-### Launch and shutdown
+### Launch and shutdown (WIP)
 
 1. Initial launch
 
     1. Download the jar file and copy into an empty folder
 
-    1. Double-click the jar file Expected: Shows the GUI with a set of sample contacts. The window size may not be optimum.
+    1. Double-click the jar file Expected: Shows the GUI with a set of sample candidates. The window size may not be optimum.
```
Comment (domlimm) [inline docs/DeveloperGuide.md:952]: Nice catch char!

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -910,13 +943,13 @@ testers are expected to do more *exploratory* testing.
 
 </div>
 
-### Launch and shutdown
+### Launch and shutdown (WIP)
```
Comment (domlimm) [inline docs/DeveloperGuide.md:946]: Yeah makes sense for WIP! :)

**Review by domlimm — COMMENTED** [review]
All LGTM @charmainehly! Nice! Perhaps please take a look at my comments. Will approve and merge after!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (charmainehly) [reply docs/DeveloperGuide.md:649]: cool, thank you! not sure if the rest are working on any parts here, maybe can do a quick check w them. I only added for the few commands I was more involved in for now 👍

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (charmainehly) [reply docs/DeveloperGuide.md:624]: hmmm am worried about users asking why there isnt a label for the module/classification of different lists for the TA candidates, so thought we should scope it down to indicate the app should be used for a specific module at any point, so it eliminates the need for the label as the whole list will be candidates applying for a specific module.

But I see where you are coming from! let me delete this!

Comment (domlimm) [reply docs/DeveloperGuide.md]: I mean you made a good point too @charmainehly now that you argued it this way... UGH.. YOU DECIDE PLEASE. can make the change whichever way @charmainehly. will wait!

Comment (charmainehly) [reply docs/DeveloperGuide.md]: It's alright! Have removed it for now :) I think the argument definitely works both ways. I can add it to the discussion points in the issue I opened up!! But I think you make sense no worries!!

**Review by domlimm — APPROVED** [review]
LGTM, merging @charmainehly! 💯

**Pull request merged**